src/isolate.rs

```diff
@@ -282,23 +289,77 @@ impl Isolate {
     }
   }
 
+  /// Runs the default MicrotaskQueue until it gets empty.
+  /// Any exceptions thrown by microtask callbacks are swallowed.
+  pub fn run_microtasks(&mut self) {
+    unsafe { v8__Isolate__RunMicrotasks(self) }
+  }
+
+  /// Enqueues the callback to the default MicrotaskQueue
+  pub fn enqueue_microtask(&mut self, microtask: Local<Function>) {
+    unsafe { v8__Isolate__EnqueueMicrotask(self, microtask) }
+  }
+
+  /// Disposes the isolate.  The isolate must not be entered by any
+  /// thread to be disposable.
+  unsafe fn dispose(&mut self) {
+    IsolateHandle::dispose(self);
+    v8__Isolate__Dispose(self)
+  }
+}
+
+// TODO(ry) Use AtomicPtr? Use Cell? Only one thread ever modifies it.
+/// Thread safe reference to an Isolate. This handle will out-live the Isolate
+/// itself, when that happens all methods on it will return false.
+pub struct IsolateHandle(Arc<Mutex<Option<*mut Isolate>>>);
```

Comment:
Why is a Mutex needed? And why care to remove Locker if you sneak a Mutex in behind the curtains?

Reply:
Not used or accessed by the the hot loops. ArcMutex is just my default too for these jobs. What would you suggest?

Reply:
* `pub struct IsolateHandle(NonNull<Isolate>);`
* create a `Box<AtomicUsize>`, cast it to `*mut AtomicUsize`, stuff that pointer in user data slot 0.